### PR TITLE
Starwalker-Class!

### PR DESCRIPTION
**Pull request by spockye**
## About The Pull Request
image

image

this pr adds a new ship, called the Starwalker class! this ship's purpose is as a small carrier used for piracy.  it comes equipped with a small hangar, two emitter bays, a well-stocked medbay, and a decently filled armoury.  

## Why It's Good For The Game
we currently only have ONE pirate ship, and i want to change that! 

## Changelog
:cl:
add: Starwalker class pirate carrier
add: a new pirate subtype of the meat hook
/:cl:

### CODE REVIEW

**Comment by PiperDoots** [comment]
Can you add a picture of the ship please

**Comment by spockye** [comment]
> Can you add a picture of the ship please

was waiting for mapdiff to finish rendering!

**Comment by Apogee-dev** [comment]
The bridge still needs a chair or two and I might suggest mixing up the leading edge of the ship a little since it’s totally flat right now. Maybe just extend the middle section of the front one tile forward from the side hangar or something. Other than that… I can’t say I’m a fan of all the tinyfan airlocks here. I dislike tinyfans in general but at the very least they should be restricted to cargo bays, not regular airlocks. If at all possible, those should be AAC airlocks instead.

**Comment by spockye** [comment]
> can’t say I’m a fan of all the tinyfan airlocks here. I dislike tinyfans in general but at the very least they should be restricted to cargo bays, not regular airlocks. If at all possible, those should be AAC airlocks instead.

theres only one tinyfan airlock, and its the one at cargo. the ones at the engineering bays dont have ones, and are boarded up.

File: code/modules/clothing/suits/miscellaneous.dm [hunk 1]
```diff
@@ -419,6 +419,15 @@
 	heat_protection = CHEST|GROIN|ARMS|LEGS
 	cold_protection = CHEST|GROIN|ARMS|LEGS
 
+/obj/item/clothing/suit/jacket/leather/duster/armoured
```
Comment (Fikou) [inline code/modules/clothing/suits/miscellaneous.dm]: this needs a different name or icon so you can tell whether its armored or not

File: code/modules/jobs/job_types/security_officer.dm [hunk 2]
```diff
@@ -281,6 +281,14 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 	id = /obj/item/card/id/syndicate_command/crew_id
 	backpack_contents = list(/obj/item/storage/box/survival/syndie=1)
 
+/datum/outfit/job/security/pirate/pilot
+	name = "First mate(pilot)"
```
Comment (Fikou) [inline code/modules/jobs/job_types/security_officer.dm]: make this normal

File: code/modules/jobs/job_types/security_officer.dm [hunk 3]
```diff
@@ -281,6 +281,14 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
...
+	uniform = /obj/item/clothing/under/costume/singer/yellow
+	shoes = /obj/item/clothing/shoes/jackboots
+	glasses = /obj/item/clothing/glasses/sunglasses/big
+	suit = /obj/item/clothing/suit/jacket/leather/duster/armoured
+	backpack_contents = list(/obj/item/storage/box/survival/security=1)
```
Comment (Fikou) [inline code/modules/jobs/job_types/security_officer.dm]: survival security box is already included by the box variable, this will make them spawn with 2

Comment (spockye) [reply code/modules/jobs/job_types/security_officer.dm]: ooh, alright. i'll get that fixed

File: code/modules/jobs/job_types/security_officer.dm (hunk 2 above)
Comment (spockye) [reply code/modules/jobs/job_types/security_officer.dm]: what do you mean with normal?

Comment (Fikou) [reply code/modules/jobs/job_types/security_officer.dm]: it does not follow the other outfits in "grammar"

File: code/modules/clothing/suits/miscellaneous.dm (hunk 1 above)
Comment (spockye) [reply code/modules/clothing/suits/miscellaneous.dm]: changed it to "Armoured leather duster"

**Comment by spockye** [comment]
if anyone could show me how to force the ship to always land in the same direction, it would be greatly appreciated. the current pod doors I'm using don't have automated directional sprites, and break upon landing. 
(it also generally just looks worse when it lands in the unintended direction)

**Review by Latentish — APPROVED** [review]
TMable? No not yet there are requests ahead of this. But it looks good so far.

The original _**Starwalker...**_
image

File: code/modules/clothing/suits/miscellaneous.dm (hunk 1 above)
Comment (Apogee-dev) [inline code/modules/clothing/suits/miscellaneous.dm]: Is this strictly necessary? We have a black armored duster already, which is a subtype of the armor vest (and therefore also benefits from the suit storage permissions that come with it). Having a brown version (admittedly with weaker armor) seems redundant. Additionally, if you do keep it, change all new instances of "armoured" to "armored" for consistency. We may as well standardize at some point and most other clothing items use the american spelling.

File: code/modules/jobs/job_types/security_officer.dm [hunk 4]
```diff
@@ -281,6 +281,13 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 	id = /obj/item/card/id/syndicate_command/crew_id
 	backpack_contents = list(/obj/item/storage/box/survival/syndie=1)
 
+/datum/outfit/job/security/pirate/pilot
+	name = "First Mate (pilot)"
+	uniform = /obj/item/clothing/under/costume/singer/yellow
```
Comment (Apogee-dev) [inline code/modules/jobs/job_types/security_officer.dm:286]: This seems like an odd choice in uniform for a normal job, considering it's a Vocaloid costume. It also has frankly ugly sprites and isn't likely to get resprited anytime soon.

File: code/modules/jobs/job_types/security_officer.dm [hunk 5]
```diff
@@ -281,6 +281,13 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 	id = /obj/item/card/id/syndicate_command/crew_id
 	backpack_contents = list(/obj/item/storage/box/survival/syndie=1)
 
+/datum/outfit/job/security/pirate/pilot
```
Comment (Apogee-dev) [inline code/modules/jobs/job_types/security_officer.dm:284]: Why is this a subtype of security officer? "first mate" positions are typically a subtype of HoP and treated as officer ranks. You might want to use the existing pirate HoP outfit for this slot.

File: _maps/configs/pirate_apostolis [hunk 6]
```diff
@@ -0,0 +1,37 @@
+{
+	"map_name": "apostolis-class carrier",
```
Comment (Apogee-dev) [inline _maps/configs/pirate_apostolis]: Minor gripe, but the name doesn't seem very fitting for a pirate ship. Greek tends to come off as having a bit more pretense than usual and pirates are anything but. I'd suggest something explicitly pirate-related like nautical terminology or the name of a historical port (Tortuga, Port Royal, and so on).

**Review by Apogee-dev — CHANGES_REQUESTED** [review]

**Comment by Apogee-dev** [comment]
Other than the issues raised in the review, layout looks good with one exception. The boarded up rear airlocks should either have tinyfans (not ideal but preferable to nothing) or be turned into AAC airlocks. In their current state they're impossible to use without leaking atmos into the engine bays (and ultimately the rest of the ship), and they're the only way out of the back of the ship so they _will_ be used.

**Comment by spockye** [comment]
for the first mate role, he doesn't act like a "hop", but more as the pod pilot. I chose his set of clothing after messing around in-game, found something that fit well together, and gave a look that I felt fitting. alone Vocaloid costume looks pretty bad, but it also looks decent as a "tattered pirate shirt" when it's under another set of clothing.  Same with the leather duster. i was thinking about using the Hos version, but it just didnt feel very piratey, unlike the leather duster. (I'll get working on Americanizing it ;()

File: _maps/configs/pirate_apostolis (hunk 6 above)
Comment (spockye) [reply _maps/configs/pirate_apostolis]: i chose the name after searching up libertia, and choosing one of its "sister names" wanting to keep the pirate ships in the same naming category. though I'll try to find a new one

**Comment by Apogee-dev** [comment]
> for the first mate role, he doesn't act like a "hop", but more as the pod pilot. I chose his set of clothing after messing around in-game, found something that fit well together, and gave a look that I felt fitting. alone Vocaloid costume looks pretty bad, but it also looks decent as a "tattered pirate shirt" when it's under another set of clothing. Same with the leather duster. i was thinking about using the Hos version, but it just didnt feel very piratey, unlike the leather duster. (I'll get working on Americanizing it ;()

If he's acting as the pilot I'd choose a different name and remove the officer flag, then. You could call the job "Flyboy" or "Pod Jockey" or something similarly informal. I'd also recommend adding a proper first-mate-as-officer role just to ensure the ship actually has two officers (including the captain). Honestly I also think you should go with the black duster or an unarmored brown duster. Snowflaking in an armored brown variant just for this one job just doesn't seem like the greatest idea to me.

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 7]
```diff
@@ -539,6 +539,16 @@
 	damage = 0
 	stamina = 40
 
+//pirate version
+/obj/item/gun/magic/hook/pirate
+	name = "boarding hook"
+	desc = "A long metal chain, with a large hook connected to the end. Once used to board ancient Merchant vessels, now used to drag un-suspecting Planet-crabs closer to you."
```
Comment (Apogee-dev) [inline code/modules/mining/lavaland/necropolis_chests.dm]: i suspect "planet-crabs" might be a direct translation or a modified idiom that didn't quite make it to english very well.  You might want to make the description a little more clearly descriptive, particularly when it comes to how this differs from a normal meat hook. Suggested desc below.

desc = "A long metal chain, with a heavy hook connected to the end that can make deadly blows in close quarters. Once used to board ancient merchant vessels, now used to drag unsuspecting victims into close range."

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 8]
```diff
@@ -539,6 +539,16 @@
...
+	ammo_type = /obj/item/ammo_casing/magic/hook/bounty
+
+/obj/item/gun/magic/hook/melee_attack_chain(mob/user, atom/target, params)
+	..()
+	user.changeNext_move(CLICK_CD_MELEE * 1)//quick to swing. 15 force can be quite something with this attack frequency.
```
Comment (Apogee-dev) [inline code/modules/mining/lavaland/necropolis_chests.dm]: the *1 is unnecessary here, multiplying by one doesn't do anything. just make it (CLICK_CD_MELEE).

File: _maps/shuttles/shiptest/pirate_starwalker.dmm [hunk 9]
```diff
@@ -0,0 +1,4077 @@
...
+	},
+/obj/item/gun/ballistic/shotgun/doublebarrel/improvised/sawn{
+	pixel_x = 5;
+	pixel_y = 2
+	},
+/obj/item/gun/ballistic/shotgun/doublebarrel/improvised/sawn{
+	pixel_y = 6
+	},
+/obj/item/gun/ballistic/automatic/zip_pistol,
+/obj/item/gun/ballistic/automatic/zip_pistol,
```
Comment (Apogee-dev) [inline _maps/shuttles/shiptest/pirate_starwalker.dmm]: suggested armory loadout: 2x pump shotgun, 2x .38 revolver, 2x winchester. Nix the zipguns and improv shotguns and replace the shivs with hunting knives. They're basically just kitchen knives that butcher faster. The crew can print kitchen knives anyway so they'll just replace the shivs immediately.

File: _maps/configs/pirate_apostolis [hunk 10]
```diff
@@ -0,0 +1,47 @@
+{
+	"map_name": "Starwalker-class carrier",
+	"prefix": "ISV",
+	"namelists": ["BEASTS", "WEAPONS"],
```
Comment (Apogee-dev) [inline _maps/configs/pirate_apostolis]: Add the pirate namelist. What's the point of a pirate ship that doesn't use the funny pirate names?

File: _maps/configs/pirate_apostolis [hunk 11]
```diff
@@ -0,0 +1,47 @@
+{
+	"map_name": "Starwalker-class carrier",
```
Comment (Apogee-dev) [inline _maps/configs/pirate_apostolis]: "Starwalker" is actually already a standard ship name in the general namelist. This may cause some confusion.

**Review by Apogee-dev — CHANGES_REQUESTED** [review]

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 8 above)
Comment (Sylphily) [reply code/modules/mining/lavaland/necropolis_chests.dm]: whoops my bad lol

File: _maps/configs/pirate_apostolis (hunk 11 above)
Comment (Latentish) [reply _maps/configs/pirate_apostolis]: I dont think this is a huge issue tbh

Comment (Apogee-dev) [reply _maps/configs/pirate_apostolis]: It's also a bit jokey and doesn't really sound appropriate/piratey, tbh, but those are personal gripes.

**Comment by Sylphily** [comment]
that outfit for that one role is a bit funky

File: _maps/configs/pirate_apostolis (hunk 10 above)
Comment (spockye) [reply _maps/configs/pirate_apostolis]: i couldnt find a pirate prefix... is it just called "pirate" ?

Comment (Apogee-dev) [reply _maps/configs/pirate_apostolis]: Namelist, not prefix. It’s the same one the Libertatia uses, “PIRATES”.

File: _maps/shuttles/shiptest/pirate_starwalker.dmm [hunk 12]
```diff
@@ -0,0 +1,4047 @@
...
+"aR" = (
+/turf/closed/wall/mineral/plastitanium,
+/area/ship/hallway/central)
+"bn" = (
+/obj/machinery/light/small{
+	dir = 1
+	},
+/obj/machinery/atmospherics/components/unary/tank/air{
+	piping_layer = 2;
+	volume = 25000
```
Comment (Apogee-dev) [inline _maps/shuttles/shiptest/pirate_starwalker.dmm]: Please don't varedit statistics like volume. If it's really necessary, you can add another air tank.

File: _maps/shuttles/shiptest/pirate_starwalker.dmm [hunk 13]
```diff
@@ -0,0 +1,4047 @@
...
+/obj/structure/railing,
+/obj/item/kitchen/knife/shiv,
+/obj/item/kitchen/knife/shiv,
+/obj/item/kitchen/knife/combat,
+/obj/effect/decal/cleanable/dirt/dust,
+/obj/item/gun/ballistic/shotgun/winchester,
+/obj/item/gun/ballistic/shotgun/winchester,
+/obj/item/gun/ballistic/revolver/detective,
+/obj/item/gun/ballistic/shotgun/riot,
+/obj/item/gun/ballistic/shotgun/riot,
```
Comment (Apogee-dev) [inline _maps/shuttles/shiptest/pirate_starwalker.dmm]: Riot shotguns might be overkill- while balance was different at the time, riot shotties were a major reason for the Libertatia's early nerfs. Also keep in mind they come pre-loaded with beanbags. Use the normal, 4-shot variety. Also, don't forget to re-add the boarding hook. You have it in the code but not placed on the ship.

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 14]
```diff
@@ -539,6 +539,16 @@
...
 
+//pirate version
+/obj/item/gun/magic/hook/pirate
+	name = "boarding hook"
+	desc = "A long metal chain, with a heavy hook connected to the end that can make deadly blows in close quarters. Once used to board ancient merchant vessels, now used to drag unsuspecting victims into close range."
+	ammo_type = /obj/item/ammo_casing/magic/hook/bounty
+
+/obj/item/gun/magic/hook/melee_attack_chain(mob/user, atom/target, params)
+	..()
+	user.changeNext_move(CLICK_CD_MELEE)//quick to swing. 15 force can be quite something with this attack frequency.
```
Comment (Apogee-dev) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Don't forget to remove the comment here, it's copied from the base meat hook and doesn't apply to the boarding hook.

**Review by Apogee-dev — CHANGES_REQUESTED** [review]

File: _maps/shuttles/shiptest/pirate_starwalker.dmm [hunk 15]
```diff
@@ -0,0 +1,4047 @@
...
+/obj/effect/decal/cleanable/dirt/dust,
+/obj/item/gun/ballistic/shotgun/winchester,
+/obj/item/gun/ballistic/shotgun/winchester,
+/obj/item/gun/ballistic/revolver/detective,
+/obj/structure/cable/orange{
+	icon_state = "2-10"
+	},
+/obj/item/gun/magic/hook/pirate,
+/obj/item/gun/ballistic/shotgun/doublebarrel,
+/obj/item/gun/ballistic/shotgun/doublebarrel,
```
Comment (Apogee-dev) [inline _maps/shuttles/shiptest/pirate_starwalker.dmm]: db shotguns swing back into not having quite enough firepower. just replace with /obj/item/gun/ballistic/shotgun/lethal. everything else is good

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 16]
```diff
@@ -539,6 +539,16 @@
...
 	stamina = 40
 
+//pirate version
+/obj/item/gun/magic/hook/pirate
+	name = "boarding hook"
+	desc = "A long metal chain, with a heavy hook connected to the end that can make deadly blows in close quarters. Once used to board ancient merchant vessels, now used to drag unsuspecting victims into close range."
+	ammo_type = /obj/item/ammo_casing/magic/hook/bounty
+
+/obj/item/gun/magic/hook/melee_attack_chain(mob/user, atom/target, params)
+	..()
```
Comment (Fikou) [inline code/modules/mining/lavaland/necropolis_chests.dm]: You need to set return value here.

File: code/modules/jobs/job_types/security_officer.dm [hunk 17]
```diff
@@ -281,6 +281,13 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
...
+	name = "Pod Jockey (pilot)"
```
Comment (Fikou) [inline code/modules/jobs/job_types/security_officer.dm]:
```suggestion
	name = "Pod Jockey (Pilot)"
```

**Review by Fikou — CHANGES_REQUESTED** [review]

**Review by Latentish — CHANGES_REQUESTED** [review]
image
You have a redundant (assumedly) copy of the ship from before you renamed it. Please remove it if it is not needed.

After that I'd like to see this get TM'd to see how it plays in game and to find more issues.

**Comment by spockye** [comment]
😔

**Comment by Latentish** [comment]
L😔

**Comment by Sylphily** [comment]
could some one explain what the fuck is failing?

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 18]
```diff
@@ -539,6 +539,17 @@
...
+/obj/item/gun/magic/hook/pirate
+	name = "boarding hook"
+	desc = "A long metal chain, with a heavy hook connected to the end that can make deadly blows in close quarters. Once used to board ancient merchant vessels, now used to drag unsuspecting victims into close range."
+	ammo_type = /obj/item/ammo_casing/magic/hook/bounty
+
+/obj/item/gun/magic/hook/pirate/melee_attack_chain(mob/user, atom/target, params)
+	. = ..()
+	user.changeNext_move(CLICK_CD_MELEE)//makes the hook unquick to swing
+	return .
+
```
Comment (ZephyrTFA) [inline code/modules/mining/lavaland/necropolis_chests.dm]:
```suggestion
/obj/item/gun/magic/hook/pirate
	name = "boarding hook"
	desc = "A long metal chain, with a heavy hook connected to the end that can make deadly blows in close quarters. Once used to board ancient merchant vessels, now used to drag unsuspecting victims into close range."
	ammo_type = /obj/item/ammo_casing/magic/hook/bounty

/obj/item/gun/magic/hook/pirate/melee_attack_chain(mob/user, atom/target, params)
	. = ..()
	user.changeNext_move(CLICK_CD_MELEE)//makes the hook unquick to swing
	return .
 
```

Comment (spockye) [reply code/modules/mining/lavaland/necropolis_chests.dm]: Something still seems to be failing...

🥲

**Comment by ZephyrTFA** [comment]
Actually reading the Linters might help you

**Comment by ZephyrTFA** [comment]
```
ERROR: found door names without proper upper-casing. 
```

**Comment by spockye** [comment]
Oh dear

**Comment by Sylphily** [comment]
lets just not do the decals for now

**Comment by spockye** [comment]
Im furious, and sad. 

I'll do this another time

**Pull request closed without merging**